src/js/actions/mask.js

```diff
@@ -34,6 +34,45 @@ define(function (require, exports) {
     var _CLEAR_PATH = 106;
 
     /**
+     * Handle escaping from vector mask for the selection based vector mask tools
+     * most tools can just call toolActions.changeVectorMaskMode directly
+     *
+     * @return {Promise}
+     */
+    var handleEscapeVectorMask = function () {
+        var appStore = this.flux.store("application"),
+            toolStore = this.flux.store("tool"),
+            currentDocument = appStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        if (toolStore.getVectorMode() && toolStore.getModalToolState()) {
+            return this.transfer(toolActions.changeVectorMaskMode, false);
+        }
+
+        var currentLayer = currentDocument.layers.selected.first();
```

Comment:
This could be null if there are no selected layers. Probably the action should short circuit somehow in this case?